editor/src/components/canvas/remix/remix-utils.tsx

```diff
@@ -543,3 +543,145 @@ export function isPageTemplate(u: unknown): u is PageTemplate {
   const maybe = u as PageTemplate
   return u != null && typeof u == 'object' && maybe.label != null && maybe.path != null
 }
+
+const AppRoutesPrefix = '/app/routes/'
+
+export function isInsideRemixFolder(filename: string): boolean {
+  return filename.startsWith(AppRoutesPrefix)
+}
+
+const possibleRemixSuffixSeparators = ['.', '_.']
+
+/**
+ * Return whether the given filename is a valid Remix route filename that is
+ * a prefix of the oldPath.
+ */
+export function remixFilenameMatchPrefix(filename: string, oldPath: string): boolean {
+  // if it's not a remix route (meaning a .jsx file inside the /app/routes/ folder), stop here
+  const isRemixRoute =
+    isInsideRemixFolder(oldPath) &&
+    isInsideRemixFolder(filename) &&
+    path.extname(filename) === '.jsx'
+  if (!isRemixRoute) {
+    return false
+  }
+
+  // to make it easier to compare paths, make them relative to the /app/routes folder and remove any optional prefixes
+  const relativeOldPath = oldPath.replace(AppRoutesPrefix, '') // without /app/routes
+  const relativeFilename = filename
+    .replace(AppRoutesPrefix, '') // without /app/routes
+    .replace(/^\([^)]+\)\./, '') // without optional prefix
+
+  return possibleRemixSuffixSeparators.some((sep) =>
+    relativeFilename.startsWith(relativeOldPath + sep),
+  )
+}
+
+export function renameRemixFile(
+  filename: string,
+  oldPath: string,
+  newPath: string,
+): {
```

Comment:
I would create a params object from this to avoid mixing up the arguments

Reply:
[`7be7bf7` (#5332)](https://github.com/concrete-utopia/utopia/pull/5332/commits/7be7bf77c4388f58b029a044ced8d666bf88122c) and bonus [`f3f40a4` (#5332)](https://github.com/concrete-utopia/utopia/pull/5332/commits/f3f40a426dd273e008924767ebdcb564fbdb7418)